apis/v1alpha1/opentelemetrycollector_types.go

```diff
@@ -177,6 +177,10 @@ type OpenTelemetryTargetAllocator struct {
 	// All CR instances which the ServiceAccount has access to will be retrieved. This includes other namespaces.
 	// +optional
 	PrometheusCR OpenTelemetryTargetAllocatorPrometheusCR `json:"prometheusCR,omitempty"`
+	// PodMonitors to be selected for target discovery.
+	PodMonitorSelector *map[string]string `json:"podMonitorSelector,omitempty"`
```

Comment:
Wouldn't `OpenTelemetryTargetAllocatorPrometheusCR` be a better fit than `OpenTelemetryTargetAllocator` as the former is used to enable the CRD functionality?